draft-ietf-quic-recovery.md

```diff
@@ -663,9 +663,11 @@ in Initial packets, or the client's estimated RTT is too small. When a
 client receives Handshake or 1-RTT packets prior to obtaining Handshake keys,
 it may assume some or all of the server's Initial packets were lost.
 
-To speed up handshake completion under these conditions, an endpoint MAY send
-a packet containing unacknowledged CRYPTO data earlier than the PTO expiry,
-subject to the address validation limits in Section 8.1 of {{QUIC-TRANSPORT}}.
+To speed up handshake completion under these conditions, an endpoint MAY, for a
+limited number of occasions per each connection, send a packet containing
+unacknowledged CRYPTO data earlier than the PTO expiry, subject to the address
+validation limits in Section 8.1 of {{QUIC-TRANSPORT}}. Doing so at most once
+for each connection will be adequate to accommodate common loss scenarios.
```

Comment:
I don't think that the packet reflection this is well-enough explained to use it here.

Reply:
I am in favor of @ianswett's suggestion.

The principles we have for preventing ping-pong are:
* do not repeatedly send an ACK-eliciting ACK
* when not receiving an ACK, rely on PTO backoff

This paragraph is talking when an endpoint MAY violate those rules. Therefore, IMO it is good to point out that risk.

Reply:
I would rather add:

> A client that does not limit the number of times it sends unacknowledged CRYPTO data in response to unauthenticated data risks creating an infinite exchange of packets.

There are other reasons why this might happen (hitting an endpoint that thinks it closed, for example.)

Reply:
In 3518d54, applied changes suggested by @ianswett to the existing sentence.
In a51ad56, added the rationale suggested by @martinthomson with tweaks. Specifically, I changed the subject to "endpoints" so that it would be generic, and used "back off", as we suggest using that for Stateless Resets, etc.

PTAL.